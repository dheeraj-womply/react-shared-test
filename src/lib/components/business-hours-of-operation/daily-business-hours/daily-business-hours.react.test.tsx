import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { DailyBusinessHours } from '.';

test('Test snapshot for Daily Business Hours component when business is open', () => {
    const onClickOpenPopover = jest.fn();
    const component = renderer.create(
        <DailyBusinessHours 
            businessDay={{ day: 'Monday', hours : [ {openTime: '07:00 AM', closeTime: '03:00 PM'} ],isClosed: false }}
            onClickOpenPopover={onClickOpenPopover}
            />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});

test('Test snapshot for Daily Business Hours component when business is closed', () => {
    const onClickOpenPopover = jest.fn();
    const component = renderer.create(
        <DailyBusinessHours 
            businessDay={{ day: 'Monday', hours : [ {openTime: null, closeTime: null} ],isClosed: true }}
            onClickOpenPopover={onClickOpenPopover}
            />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});

test('Test snapshot for Daily Business Hours component when business is open but no time selected', () => {
    const onClickOpenPopover = jest.fn();
    const component = renderer.create(
        <DailyBusinessHours 
            businessDay={{ day: 'Monday', hours : [ {openTime: null, closeTime: null} ],isClosed: false }}
            onClickOpenPopover={onClickOpenPopover}
            />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});


test('Check if popover open function is called', async () => {
    const onClickOpenPopover = jest.fn();
    const { findByTestId } = render(
        <DailyBusinessHours 
            businessDay={{ day: 'Monday', hours : [ {openTime: '07:00 AM', closeTime: '03:00 PM'} ],isClosed: false }}
            onClickOpenPopover={onClickOpenPopover}
            />
    );
    fireEvent.click(await findByTestId('edit-hours'));
    expect(onClickOpenPopover).toBeCalled()
});


test('Check if onClick fn is called when disbaled prop is passed', async () => {
    const onClickOpenPopover = jest.fn();
    const onClickFn = jest.fn();

    const { findByTestId } = render(
        <DailyBusinessHours 
            disabled={true}
            onClick={onClickFn}
            businessDay={{ day: 'Monday', hours : [ {openTime: '07:00 AM', closeTime: '03:00 PM'} ],isClosed: false }}
            onClickOpenPopover={onClickOpenPopover}
            />
    );
    fireEvent.click(await findByTestId('edit-hours'));
    expect(onClickFn).toBeCalled()
});