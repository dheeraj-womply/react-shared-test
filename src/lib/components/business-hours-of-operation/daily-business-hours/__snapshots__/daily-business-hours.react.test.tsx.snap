// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot for Daily Business Hours component when business is closed 1`] = `
<div
  className="MuiGrid-root sc-AxirZ lgjYw MuiGrid-container"
>
  <div
    className="MuiGrid-root sc-AxhCb eUHVRl MuiGrid-item MuiGrid-grid-xs-4"
    color="rgba(0, 0, 0, 0.87)"
    textalign="left"
  >
    Monday
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root sc-AxhCb ihfzax MuiGrid-item MuiGrid-grid-xs-10"
        color="rgba(0, 0, 0, 0.87)"
        textalign="center"
      >
        Closed
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root sc-AxiKw cA-dXMf"
          customcolor="rgba(0, 0, 0, 0.87)"
          data-testid="edit-hours"
          float="none"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test snapshot for Daily Business Hours component when business is open 1`] = `
<div
  className="MuiGrid-root sc-AxirZ lgjYw MuiGrid-container"
>
  <div
    className="MuiGrid-root sc-AxhCb eUHVRl MuiGrid-item MuiGrid-grid-xs-4"
    color="rgba(0, 0, 0, 0.87)"
    textalign="left"
  >
    Monday
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root sc-AxhCb ihfzax MuiGrid-item MuiGrid-grid-xs-12"
            color="rgba(0, 0, 0, 0.87)"
            textalign="center"
          >
            07:00 AM - 03:00 PM
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root sc-AxiKw cA-dXMf"
          customcolor="rgba(0, 0, 0, 0.87)"
          data-testid="edit-hours"
          float="none"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test snapshot for Daily Business Hours component when business is open but no time selected 1`] = `
<div
  className="MuiGrid-root sc-AxirZ lgjYw MuiGrid-container"
>
  <div
    className="MuiGrid-root sc-AxhCb eUHVRl MuiGrid-item MuiGrid-grid-xs-4"
    color="rgba(0, 0, 0, 0.87)"
    textalign="left"
  >
    Monday
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root sc-AxhCb ihfzax MuiGrid-item MuiGrid-grid-xs-12"
            color="rgba(0, 0, 0, 0.87)"
            textalign="center"
          >
             - 
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root sc-AxiKw cA-dXMf"
          customcolor="rgba(0, 0, 0, 0.87)"
          data-testid="edit-hours"
          float="none"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
